Please update io-extras dependency to support older rust versions.

Hi, i'm trying to build code (specifically i'm trying to run the tests of the selinux crate) that uses socketpair using rustc 1.62, however when I do I get a failure in the io-extras crate.

    error[E0277]: the trait bound `T: FromFd` is not satisfied

This is fixed in io-extras 0.17, is there any chance you could update the dependency and make a new release?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -13,7 +13,7 @@ exclude = ["/.github"]
 [dependencies]
 async-std = { version = "1.10.0", optional = true }
 tokio = { version = "1.8.1", optional = true }
-io-extras = "0.16.0"
+io-extras = "0.17.0"
 io-lifetimes = { version = "1.0.0", default-features = false }
 
 [target.'cfg(not(windows))'.dependencies]

--- a/src/windows.rs
+++ b/src/windows.rs
@@ -2,8 +2,7 @@
 
 use io_extras::os::windows::{
     AsHandleOrSocket, AsRawHandleOrSocket, AsRawReadWriteHandleOrSocket, AsReadWriteHandleOrSocket,
-    BorrowedHandleOrSocket, IntoHandleOrSocket, IntoRawHandleOrSocket, OwnedHandleOrSocket,
-    RawHandleOrSocket,
+    BorrowedHandleOrSocket, IntoRawHandleOrSocket, OwnedHandleOrSocket, RawHandleOrSocket,
 };
 use io_lifetimes::{AsHandle, BorrowedHandle, OwnedHandle};
 use std::cmp::min;
@@ -304,10 +303,10 @@ impl IntoRawHandleOrSocket for SocketpairStream {
     }
 }
 
-impl IntoHandleOrSocket for SocketpairStream {
+impl From<SocketpairStream> for OwnedHandleOrSocket {
     #[inline]
-    fn into_handle_or_socket(self) -> OwnedHandleOrSocket {
-        self.0.into_handle_or_socket()
+    fn from(stream: SocketpairStream) -> Self {
+        stream.0.into()
     }
 }
 

--- a/src/windows_async_std.rs
+++ b/src/windows_async_std.rs
@@ -6,10 +6,10 @@ use async_std::os::windows::io::{AsRawHandle, FromRawHandle, IntoRawHandle, RawH
 use async_std::path::Path;
 use io_extras::os::windows::{
     AsHandleOrSocket, AsRawHandleOrSocket, AsRawReadWriteHandleOrSocket, AsReadWriteHandleOrSocket,
-    BorrowedHandleOrSocket, IntoHandleOrSocket, IntoRawHandleOrSocket, OwnedHandleOrSocket,
+    BorrowedHandleOrSocket, IntoRawHandleOrSocket, OwnedHandleOrSocket,
     RawHandleOrSocket,
 };
-use io_lifetimes::{AsHandle, BorrowedHandle, IntoHandle, OwnedHandle};
+use io_lifetimes::{AsHandle, BorrowedHandle, OwnedHandle};
 use std::convert::TryInto;
 use std::fmt::{self, Debug};
 use std::os::windows::ffi::OsStrExt;
@@ -172,10 +172,10 @@ impl IntoRawHandle for AsyncStdSocketpairStream {
     }
 }
 
-impl IntoHandle for AsyncStdSocketpairStream {
+impl From<AsyncStdSocketpairStream> for OwnedHandle {
     #[inline]
-    fn into_handle(self) -> OwnedHandle {
-        self.0.into_handle()
+    fn from(stream: AsyncStdSocketpairStream) -> Self {
+        stream.0.into()
     }
 }
 
@@ -207,10 +207,10 @@ impl IntoRawHandleOrSocket for AsyncStdSocketpairStream {
     }
 }
 
-impl IntoHandleOrSocket for AsyncStdSocketpairStream {
+impl From<AsyncStdSocketpairStream> for OwnedHandleOrSocket {
     #[inline]
-    fn into_handle_or_socket(self) -> OwnedHandleOrSocket {
-        self.0.into_handle_or_socket()
+    fn from(stream: AsyncStdSocketpairStream) -> Self {
+        stream.0.into()
     }
 }
 

